### PR TITLE
Update Artemis4J

### LINKED ISSUE #44: Update to new version of Artemis4J
@Luro02 @Feuermagier would be nice, if you could take a look here. I've granted you access to the repository.

### PR DESCRIPTION
**Pull request by Luro02**
This PR closes #44.

Open issues:
- [x] merge my forks Artemis4J changes and release them https://github.com/kit-sdq/artemis4j/pull/92
- [x] switch from jitpack to an actual dependency
- [x] I removed the `|| ..isTest()` condition in `PrefixMatcher` because I did not know the equivalent code in the new API. Is this okay?
- [x] the above two points are as TODOs in the code

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -15,13 +15,17 @@
...
       <artifactId>artemis4j</artifactId>
-      <version>6.7.5</version>
+      <version>f5ad9e57a7</version>
     </dependency>
+    <!--<dependency>
+      <groupId>edu.kit.kastel.sdq</groupId>
+      <artifactId>artemis4j</artifactId>
+      <version>7.4.0-SNAPSHOT</version>
+    </dependency>-->
     <dependency>
```
Comment (dfuchss) [inline pom.xml:29]: Just that we don't forget that later :)

**Comment by dfuchss** [comment]
We should split the PR into formatting and updating. Otherwise, the review will be hard :)

File: src/main/java/edu/kit/kastel/sdq/scorestats/cli/CLI.java [hunk 2]
```diff
@@ -158,4 +159,32 @@ public void run(String[] args) {
...
+
+        for (ProgrammingSubmission submission : submissions) {
+            String studentId = submission.getParticipantIdentifier();
+            Assessment assessment = submission.openAssessment(config).orElse(null);
+            if (assessment == null) {
+                skippedStudents.add(studentId);
+                continue;
+            }
+
+            if (assessments.containsKey(studentId)) {
```
Comment (dfuchss) [inline src/main/java/edu/kit/kastel/sdq/scorestats/cli/CLI.java:181]: If you load two correction rounds I guess this breaks.

File: src/main/java/edu/kit/kastel/sdq/scorestats/core/assessment/PrefixMatcher.java [hunk 3]
```diff
@@ -17,18 +17,19 @@ public PrefixMatcher(String regex) {
...
      *
-     * @param feedback the feedback
-     * @return {@code true} if {@link Feedback#getTestName()} begins with the prefix
+     * @param testResult the test result
+     * @return {@code true} if {@link TestResult#getTestName()} begins with the
+     *         prefix
      */
     @Override
-    public boolean matches(Feedback feedback) {
-        if (feedback.getFeedbackType() == FeedbackType.MANUAL_UNREFERENCED || !feedback.isTest()) {
```
Comment (dfuchss) [inline src/main/java/edu/kit/kastel/sdq/scorestats/core/assessment/PrefixMatcher.java:28]: The remove of isTest should be ok as it should never occur :)

**Review by dfuchss — COMMENTED** [review]
I took a look. Just some minor comments :)

File: src/main/java/edu/kit/kastel/sdq/scorestats/cli/CLI.java (hunk 2 above)
Comment (Luro02) [reply src/main/java/edu/kit/kastel/sdq/scorestats/cli/CLI.java:181]: It seemed to work in the old code: https://github.com/kit-sdq/programming-lecture-artemis-score-stats/blob/c7f6781b8b436e1d3cc526986d70f09e591f322e/src/main/java/edu/kit/kastel/sdq/scorestats/core/client/Artemis4JArtemisClient.java#L94

Comment (dfuchss) [reply src/main/java/edu/kit/kastel/sdq/scorestats/cli/CLI.java:181]: Ok .. then. Did you tested the code for the test system or on production for your students?

Comment (Luro02) [reply src/main/java/edu/kit/kastel/sdq/scorestats/cli/CLI.java:181]: I only tested it in production

Comment (dfuchss) [reply src/main/java/edu/kit/kastel/sdq/scorestats/cli/CLI.java:181]: Perfect. I'll test it as well tomorrow :)

Comment (dfuchss) [reply src/main/java/edu/kit/kastel/sdq/scorestats/cli/CLI.java:181]: Especially Ill test it with exams .. they have two correction rounds :D

**Review by tsaglam — APPROVED** [review]
Tested for sheet 5, seems to work and produces results that look as expected.

**Review by dfuchss — APPROVED** [review]

**Pull request merged**